lib/server_sent_event_stage.ex

```diff
@@ -38,13 +41,24 @@ defmodule ServerSentEventStage do
   end
 
   # Server functions
-  defstruct [:url, :headers, :connected_url, :conn, :ref, buffer: "", redirecting?: false]
+  defstruct [
+    :url,
+    :headers,
+    :connected_url,
+    :conn,
+    :ref,
+    :idle_timeout,
+    :idle_timer,
+    buffer: "",
+    redirecting?: false
+  ]
 
   @doc false
   def init(args) do
     state = %__MODULE__{
       url: Keyword.fetch!(args, :url),
-      headers: Keyword.get(args, :headers, [])
+      headers: Keyword.get(args, :headers, []),
+      idle_timeout: Keyword.get(args, :idle_timeout, nil)
```

Comment:
question: do you think we should have a default idle timeout? [WhatWG](https://html.spec.whatwg.org/multipage/server-sent-events.html#authoring-notes) suggests that producers send a message every 15 seconds, and our current [implementation](https://github.com/mbta/api/blob/219bf5893c3ec5bab96212664423e400fd9b2373/apps/api_web/lib/api_web/event_stream.ex#L34) in the API sends a message every 30 seconds. Maybe 60 seconds, or 300 (5 minutes) would be safe defaults?

Reply:
I don't think we particularly need a default - anything generous enough to reduce the risk of surprise incompatibilities would need overriding in all of our applications anyway, at which point there may as well be no default at all.